### PR TITLE
1.13: AnvilDamageEvent

### PR DESCRIPTION
**Pull request by BillyGalbreath**
Replaces PR #981 for 1.13

### CODE REVIEW

File: Spigot-Server-Patches/0299-AnvilDamageEvent.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                     if (!entityhuman.abilities.canInstantlyBuild && iblockdata.a(TagsBlock.x) && entityhuman.getRandom().nextFloat() < 0.12F) {
+                         IBlockData iblockdata1 = BlockAnvil.a_(iblockdata);
++                        // Paper start
++                        com.destroystokyo.paper.event.block.AnvilDamagedEvent event = new com.destroystokyo.paper.event.block.AnvilDamagedEvent(getBukkitView(), org.bukkit.craftbukkit.block.data.CraftBlockData.fromData(iblockdata1));
++                        if (!event.callEvent()) {
++                            return itemstack;
++                        } else if (event.getDamageState() == com.destroystokyo.paper.event.block.AnvilDamagedEvent.DamageState.BROKEN) {
++                            iblockdata1 = null;
++                        } else {
++                            iblockdata1 = ((org.bukkit.craftbukkit.block.data.CraftBlockData) event.getDamageState().getMaterial().createBlockData()).getState();
```
Comment (aikar) [inline Spigot-Server-Patches/0299-AnvilDamageEvent.patch]: This is going to cause it to lose its facing data. We need to carry over previous facing data, see BlockAnvil.a_()

**Review by aikar — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0299-AnvilDamageEvent.patch (hunk 1 above)
Comment (BillyGalbreath) [reply Spigot-Server-Patches/0299-AnvilDamageEvent.patch]: Fixed.

**Review by aikar — APPROVED** [review]

**Comment by aikar** [comment]
Can you merge upstream and rebuild patches, CI tests appear to be failing.

**Comment by BillyGalbreath** [comment]
There we go. That was weird.

**Pull request merged**